google/resumable_media/requests/download.py

```diff
@@ -58,12 +62,43 @@ def _write_to_stream(self, response):
 
         Args:
             response (~requests.Response): The HTTP response object.
+
+        Raises:
+            DataCorruption: If the download's checksum doesn't agree with
+                server-computed checksum.
         """
+        md5_hash = hashlib.md5()
+        expected_md5_hash = None
+        if ('X-Goog-Hash' in response.headers
+            and response.headers['X-Goog-Hash']):
+            for checksum in response.headers['X-Goog-Hash'].split(','):
+                name, value = checksum.split('=', 1)
+                if name == 'md5':
+                    expected_md5_hash = value
+        if not expected_md5_hash:
+            self._logger.info(
+                'No MD5 checksum was returned from the service while '
+                'downloading %s (which happens for composite objects), so '
+                'client-side content integrity checking is not being '
+                'performed.' % self.media_url)
         with response:
             body_iter = response.iter_content(
                 chunk_size=_SINGLE_GET_CHUNK_SIZE, decode_unicode=False)
             for chunk in body_iter:
                 self._stream.write(chunk)
+                md5_hash.update(chunk)
+        if sys.version_info[:3] < (3,):
+            actual_md5_hash = (base64.encodestring(md5_hash.digest())
+                               .rstrip('\n'))
+        else:
+            actual_md5_hash = (base64.encodebytes(bytes(md5_hash.digest(),
+                                                        'UTF-8')).rstrip('\n'))
```

Comment:
FYI I added this conditional logic because of the test failure at https://circleci.com/gh/GoogleCloudPlatform/google-resumable-media-python/195#build-timing/containers/0
(I am unable to install Python3.6 on my machine, so Luke suggested just testing under Python 2.7 and letting Travis CI do the 3.6 tests...)